internal/server/multitherading.go

```diff
@@ -0,0 +1,108 @@
+package server
+
+import (
+	"bytes"
+
+	"github.com/dicedb/dice/internal/cmd"
+	"github.com/dicedb/dice/internal/comm"
+	"github.com/dicedb/dice/internal/eval"
+	"github.com/dicedb/dice/internal/ops"
+	"github.com/dicedb/dice/internal/shard"
+	"github.com/twmb/murmur3"
+)
+
+func getShard(key string, n uint32) uint32 {
+	hash := murmur3.Sum32([]byte(key))
+	return hash % n
+}
+
+func (s *AsyncServer) cmdsBreakup(redisCmd *cmd.RedisCmd, c *comm.Client) []cmd.RedisCmd {
+	val, ok := WorkerCmdsMeta[redisCmd.Cmd]
+
+	if !ok {
+		return []cmd.RedisCmd{*redisCmd}
+	}
+
+	// if command supports multisharding then send the command
+	// to the respective breakup function
+	// Which can return array of broken down commands
+	return val.Breakup(s.shardManager, redisCmd, c)
+}
+
+func (s *AsyncServer) scatter(cmds []cmd.RedisCmd, c *comm.Client) {
```

Comment:
This logic should change right and distribute based on CmdType. Correct?

Reply:
@pratikpandey21 the logic is before scatter call, https://github.com/DiceDB/dice/pull/634/files#diff-d0706a4ae4ecf91c5224ca388f31a087a36af06c40dc81321da7958b25c70b23R276. Still do you think we need to change shard scatter logic?